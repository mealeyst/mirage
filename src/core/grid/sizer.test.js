import React from 'react'
import 'jest-styled-components'

import { Sizer } from 'SRC'

const { shallowWithTheme } = global

describe('(Styled Component) Sizer', () => {
  const createSizer = (props) => {
    return shallowWithTheme(<Sizer {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createSizer())
    .toMatchSnapshot()
  })

  test('grid-column-end should be set', () => {
    expect(createSizer({desktop: {width: 4}}))
    .toHaveStyleRule({'grid-column-end': 'span 4'})
  })
})
