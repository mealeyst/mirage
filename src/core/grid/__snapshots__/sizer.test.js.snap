// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Styled Component) Sizer matching the snapshot 1`] = `
.c0 {
  width: calc((100% * (4/4)) - 2rem);
  box-sizing: border-box;
  margin: 0 1rem;
}

@supports (display:grid) {
  .c0 {
    width: auto;
    margin: 0;
    grid-column-end: span 4;
  }
}

<div
  className="c0"
/>
`;
